### PR TITLE
[RSDK-8923] Acquire app address from cloud credentials

### PR DESCRIPTION
**Pull request by gvaradarajan**
Instead of hardcoding the signaling address to http://app.viam.com:443, we now use the "app_address" key given in cloud credentials

IMPORTANT NOTE: As a follow-up, the logic in the project template under the "has_robot_config" condition must be updated upon version bump!

### CODE REVIEW

File: templates/project/src/main.rs [hunk 1]
```diff
@@ -80,6 +81,7 @@ fn main() {
 
     if !storage.has_robot_configuration() {
         // check if any were statically compiled
+        // TODO: update with app address storage logic when version is incremented
```
Comment (acmorrow) [inline templates/project/src/main.rs:84]: I'd suggest making a ticket with a fixVersion and adding the ticket into the TODO here as `TODO(RSDK-WXYZ): Update ...`.

**Review by acmorrow — APPROVED** [review]
LGTM with a few small comments / suggestions

File: micro-rdk/src/common/credentials_storage.rs [hunk 2]
```diff
@@ -29,6 +33,10 @@ impl RobotCredentials {
     pub(crate) fn robot_secret(&self) -> &str {
         &self.robot_secret
     }
+
+    pub(crate) fn app_address(&self) -> Uri {
+        self.app_address.parse::<Uri>().unwrap()
```
[the remark this answers is not in the document]
Comment (gvaradarajan) [reply micro-rdk/src/common/credentials_storage.rs:38]: There's no good reason, I'll change it accordingly

File: templates/project/src/main.rs (hunk 1 above)
Comment (gvaradarajan) [reply templates/project/src/main.rs:84]: will do

File: micro-rdk/src/common/conn/viam.rs [hunk 3]
```diff
@@ -1403,7 +1394,7 @@ mod tests {
         let creds = CloudConfig {
             id: "".to_string(),
             secret: "".to_string(),
-            app_address: "".to_owned(),
+            app_address: "http://localhost:56563".to_owned(),
```
Comment (mattjperez) [inline micro-rdk/src/common/conn/viam.rs:1397]: lets make this a local const since it's referenced so often

**Review by mattjperez — APPROVED** [review]
LGTM, just a suggestion

**Pull request merged**